frontend/src/components/applyForm/README.md

```diff
@@ -0,0 +1,78 @@
+# Apply form component
+
+This component takes form and UI schemas to create a form. The form schema conforms to JSON Schema and can take any shape as long as it conforms to JSON Schema.
+
+[AjV](https://ajv.js.org) is used to validate the schema when saved.
+
+The UI schema is custom for this project. It was inspired by [react-jsonschema-form](https://github.com/rjsf-team/react-jsonschema-form). RJSF was not chosen because it doesn't work with server actions, it doesn't support hierarchical UI layouts natively, and the validation will move to the API which reduces the value of the tightly coupled validation in RJSF. There is a [plugin for layouts](https://github.com/audibene-labs/react-jsonschema-form-layout), however it hasn't been updated in 4 years and still requires a lot of abstraction. This component provides most of the UI rendering with a ~100 line recursive function and uses the same library for validation. The widgets however are patterned after the RJSF widgets.
+
+## UI Schema
+
+The UI Schema is an array of fields or sections.
+
+### Fields
+
+Fields can either reference a field in the form schema:
+
+```
+{
+    type: "field",
+    definition: "/properties/TestField",
```

Comment:
looking at the example schema from the DB I think this makes sense - since all of the individual fields are nested under `properties`, this is pointing to something like the field defined as "TestField" here?
```
  "properties": {
    "TestField": {
      "type": "string",
      "title": "Date of application ",
      "format": "date"
    },
```
It may help to provide an example with a bit more context like this to show this would be used

Reply:
I added this example. Additionally tried to clarify:

> A field can have a `schema` without a `definition`. However, this field will only exist in the UI. When the form is saved the field will not be included in the form data sent to the server.